timeseries/src/autogluon/timeseries/splitter.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from typing import Tuple
+
+import pandas as pd
+
+from .dataset import TimeSeriesDataFrame
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractTimeSeriesSplitter:
+    """A class that handles train / validation splitting of timeseries datasets."""
+
+    def split(
+        self, ts_dataframe: TimeSeriesDataFrame, prediction_length: int
+    ) -> Tuple[TimeSeriesDataFrame, TimeSeriesDataFrame]:
+        """Split each series in the input dataset into one train and potentially multiple validation series.
+
+        The `item_id` of each validation series is a string of format `f"{idx}_[{start}:{end}]". Here `idx` is
+        the `item_id` of the input series, and `start` and `end` correspond to the slice parameters used to generate
+        the validation series.
+
+        For example, the series `val_data.loc["0_[None:-10]"]` in the validation dataset can equivalently be
+        obtained as `ts_dataframe.loc[0][None:-10]`.
+
+        Parameters
+        ----------
+        ts_dataframe: TimeSeriesDataFrame
+            Dataset containing of series that should be split.
+        prediction_length: int
+            The forecast horizon, i.e., how many time points into the future forecasters should be trained to predict.
+
+        Returns
+        -------
+        train_data: TimeSeriesDataFrame
+            Time series used for training. Has the same number of items as in the input dataset.
+        val_data: TimeSeriesDataFrame
+            Time series used for tuning / validation.
+        """
+        return self._split(ts_dataframe=ts_dataframe, prediction_length=prediction_length)
+
+    def _split(
+        self, ts_dataframe: TimeSeriesDataFrame, prediction_length: int
+    ) -> Tuple[TimeSeriesDataFrame, TimeSeriesDataFrame]:
+        raise NotImplementedError
+
+    def describe_validation_strategy(self, prediction_length: int) -> None:
+        raise NotImplementedError
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+
+def append_suffix_to_item_id(ts_dataframe: TimeSeriesDataFrame, suffix: str) -> TimeSeriesDataFrame:
+    """Append a suffix to each item_id in a TimeSeriesDataFrame."""
+
+    def add_suffix(multiindex_element):
+        item_id, timestamp = multiindex_element
+        return (f"{item_id}{suffix}", timestamp)
+
+    result = ts_dataframe.copy(deep=False)
```

Comment:
Something bad may happen if a model tries to modify the data in-place. Should we 
1. hope that this never happens, or
2. be extra safe, do a deep copy and waste memory?